### PR TITLE
Support testing, building and pushing container for C9S

### PR DESCRIPTION
**Pull request by phracek**
This pull request adds support for building, testing and pushing container `varnish-6` for CentOS Stream 9.

What is changed in Dockerfile.c9s?
See diff:
```bash
$ diff 6/Dockerfile.rhel8 6/Dockerfile.c9s
1c1
< FROM ubi8/s2i-core:1
---
> FROM quay.io/sclorg/s2i-core-c9s:c9s
23,24c23,24
<       usage="s2i build https://github.com/sclorg/varnish-container.git --context-dir=6/test/test-app/ rhel8/varnish-6 sample-server" \
<       name="rhel8/varnish-6" \
---
>       usage="s2i build https://github.com/sclorg/varnish-container.git --context-dir=6/test/test-app/ quay.io/sclorg/varnish-6-c9s sample-server" \
>       name="sclorg/varnish-6-c9s" \
28d27
<     yum -y module enable varnish:6  && \
```

### CODE REVIEW

**Comment by phracek** [comment]
[test-all]

**Comment by phracek** [comment]
[test-all]

**Comment by phracek** [comment]
[test-all]

**Comment by phracek** [comment]
[test-all]

**Comment by phracek** [comment]
[test-openshift]

File: .github/workflows/build-and-push.yml [hunk 1]
```diff
@@ -35,41 +42,65 @@ jobs:
           ver="${{ matrix.version }}"
           echo "::set-output name=SHORT_VER::${ver//./}"
 
-      - name: Check if dockerfile is present in version directory
-        id: check_dockerfile_file
+      - name: Login to Quay.io private registry
```
Comment (zmiklank) [inline .github/workflows/build-and-push.yml:45]: Shouldn't this step be just before the push-to-quay step, and not here? 
That way, when the build fails, the quay.io would not be to be contacted at all.

**Review by zmiklank — CHANGES_REQUESTED** [review]
Thank you for the PR and for the detailed PR description. The diffs are a great way for expressing the changes!

File: .github/workflows/build-and-push.yml (hunk 1 above)
Comment (phracek) [reply .github/workflows/build-and-push.yml:45]: Good, point. I wanted to have at the beginning. When the login is called, then it is valid for action.

I wanted to have a setup for the action `build-and-push` at the beginning.

Comment (zmiklank) [reply .github/workflows/build-and-push.yml:45]: It's true, that from the past can be seen, that login to quay.io is more error-prone than building the containers.

File: .github/workflows/build-and-push.yml [hunk 2]
```diff
@@ -35,41 +42,65 @@ jobs:
...
         with:
           image: ${{ steps.build-image.outputs.image }}
           tags: ${{ steps.build-image.outputs.tags }}
-          registry: quay.io/centos7
+          registry: quay.io/${{ matrix.registry_namespace }}
           username: ${{ secrets.QUAY_IMAGE_BUILDER_USERNAME }}
           password: ${{ secrets.QUAY_IMAGE_BUILDER_TOKEN }}
 
+      - name: Push image to Quay.io/${{ matrix.registry_namespace }} namespace
+        if: steps.check_exclude_centos7_file.outputs.files_exists == 'false' && ${{ matrix.registry_namespace != 'centos7' }}
```
Comment (zmiklank) [inline .github/workflows/build-and-push.yml:90]: Aren't these braces misplaced, same as here: https://github.com/sclorg/nginx-container/pull/169/commits/1db7d3e526213916732c0c6d6a6ef4d3eb5f8d4a ?

Comment (phracek) [reply .github/workflows/build-and-push.yml:90]: Yes, the same error, I am gonna to fix them in all containers.

**Review by zmiklank — APPROVED** [review]
LGTM, thanks for the changes.

**Pull request merged**